src/main/java/org/opentripplanner/middleware/tripMonitor/jobs/CheckMonitoredTrip.java

```diff
@@ -257,133 +263,199 @@ private void enqueueNotification(TripMonitorNotification ...tripMonitorNotificat
             }
-            // If the minutes until the trip is within the lead time, check the trip every minute
-            // (assuming the loop runs every minute).
-            // TODO: Should lead time just be used for notification time limit?
-            if (minutesUntilTrip <= trip.leadTimeInMinutes) {
+            // If the trip starts within 30 minutes, check the trip every minute (assuming the loop runs every minute).
+            if (minutesUntilTrip <= 30) {
                 // TODO: Change log level.
-                LOG.info(
-                    "Trip {} happening within user's lead time ({} minutes). Checking every minute.",
-                    trip.id,
-                    trip.leadTimeInMinutes
-                );
+                LOG.info("Trip happening within 30 minutes. Checking every minute.");
                 return false;
             }
         }
         // TODO: Check that journey state is not flagged
         // TODO: Check last notification time.
         // Default to skipping check.
         // TODO: Change log level.
-        LOG.info("Trip {} criteria not met to check. Skipping.", trip.id);
+        LOG.info("Trip criteria not met to check. Skipping.");
         return true;
     }
 
     /**
-     * Generate the appropriate OTP query params for the trip for the current check. This currently involves replacing
+     * Calculate the next possible time that the itinerary is possible, advancing the targetZonedDateTime and setting
+     * the targetZonedDateTime, targetDate, otpResponse and matchingItinerary. Returns false if an error occurs or a
+     * matching itinerary isn't found on the next possible date the trip should be monitored on.
+     */
+    private boolean calculateNextItinerary() {
+        // if the trip is not active on the current zoned date time, advance until a day is found when the trip is
+        // active. It is guaranteed that the trip will be active on a certain date because a call to
+        // {@link MonitoredTrip#isInactive} should already have been made in the shouldSkipMonitoredTripCheck method.
+        while (!trip.isActiveOnDate(targetZonedDateTime)) {
+            targetZonedDateTime = targetZonedDateTime.plusDays(1);
+        }
+        targetDate = targetZonedDateTime.format(DATE_FORMATTER);
+
+        // execute trip plan request for the target time
+        LOG.info("Attempting to calculate next trip on target date {}", targetDate);
+        if (!calculateOtpResponse()) {
+            // failed to calculate, return false
+            return false;
+        }
+
+        // check if the resulting itinerary has a matching itinerary
+        if (!updateMatchingItinerary()) {
+            // TODO: figure out how to notify the user that the itinerary is not possible anymore
+            // for now, just say the next itinerary couldn't be found
+            return false;
+        }
+        return true;
+    }
+
+    private boolean calculateOtpResponse() {
```

Comment:
javadoc needed

Reply:
I feel like this method is useful to have here since it relies on a number of instance variables. I added JavaDoc in 6fd00a976c063c5b46e7254a995fbbb628b90445